lib/xebow/excalibur/keyboard.ex

```diff
@@ -0,0 +1,284 @@
+require Logger
```

Comment:
I was writing up a big explanation of why I do this, and decided to check myself, and cannot find any supporting evidence of my previous assumptions... :rofl: 

I will change this and stop doing this pattern.

Reply:
I'm honestly surprised it works this way. I thought `require` enabled macros in the required module to work in the receiving module. Now this gives me something to research. :grin:

Reply:
`require` is lexically scoped, just like alias and import.  I tried it in an iex shell and was also surprised.  So in this case, the require would be valid for all modules defined in this file.  If it were inside the module and there were a second module, then the second module would not have access to the macros.

As I said, the point is moot because we don't put multiple (top-level) modules in files by convention.